telegram/ext/commandhandler.py

```diff
@@ -135,21 +145,21 @@ def check_update(self, update):
         """
         if (isinstance(update, Update) and
                 (update.message or update.edited_message and self.allow_edited)):
-            message = update.message or update.edited_message
+            message = update.effective_message
 
-            if message.text and message.text.startswith('/') and len(message.text) > 1:
-                first_word = message.text_html.split(None, 1)[0]
-                if len(first_word) > 1 and first_word.startswith('/'):
-                    command = first_word[1:].split('@')
-                    command.append(
-                        message.bot.username)  # in case the command was sent without a username
+            if message.entities and message.entities[0].type == MessageEntity.BOT_COMMAND and \
+                    message.entities[0].offset == 0:
+                command = message.text[1:message.entities[0].length]
```

Comment:
I know this technically works because command can only be ascii, but maybe using the parse entity method would be a good idea here

Reply:
I think it's easiest and safest this way.
If we used `parse_entities` we would have to search the dict for either message.text.split()[0] or check every entity for it's offset. Dicts are not ordered on <py3.5.
Since as you pointed out the entity is ascii and we know for sure it should be the first one in `message.entites` I think this is the best way.